src/pages/PublishCar/FormSelectInput.component.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { UseFormRegisterReturn } from "react-hook-form";
+import styled from "styled-components";
+
+import { InputStyle } from "../../components/styled";
+import { defaultTheme } from "../../theme";
+
+type MapElement = { id: number; name: string };
+interface State extends MapElement {
+  cities: Array<MapElement>;
+}
+interface Props {
+  label: string;
+  register: UseFormRegisterReturn;
+  loading: boolean;
+  onFocus?: React.FocusEventHandler<HTMLSelectElement>;
+  onBlur?: React.FocusEventHandler<HTMLSelectElement>;
+  onChange?: (e: React.ChangeEvent) => void;
+  data?: Array<any>;
+  disabled?: boolean;
+}
+export const FormSelectInput = ({
+  label,
+  register,
+  onFocus,
+  onBlur,
+  onChange,
+  loading,
+  data,
+  disabled,
+}: Props) => {
+  let selectBody: React.ReactElement;
+
+  if (loading) selectBody = <option value="">Loading..</option>;
+
+  if (label === "Cities")
+    selectBody = (
+      <>
+        <option value="">{"Select an option"}</option>
+        {data?.map((state: State) => (
+          <optgroup key={state.id} label={state.name}>
+            {state.cities.map((city: MapElement) => (
+              <option key={city.id} value={city.id}>
+                {city.name}
+              </option>
+            ))}
+          </optgroup>
+        ))}
+      </>
+    );
```

Comment:
I'm skeptical that this should not be here. The problem with something like this is it will make the component grow indefinitely as more and more use cases come along. You want to build these things in a way that it can handle anything data. Making it truly reusable. An example of a possible solution would be to pass in a `renderOption` prop, which is a function, that you call on every piece of data.

```
      <>
        <option value="">{"Select an option"}</option>
        {data?.map(props.renderOption)}
      </>

....

<FormSelectInput data={[]} renderOption={(state: State) => (
          <optgroup key={state.id} label={state.name}>
            {state.cities.map((city: MapElement) => (
              <option key={city.id} value={city.id}>
                {city.name}
              </option>
            ))}
          </optgroup>
        )}
}
```